onnxruntime/python/tools/symbolic_shape_infer.py

```diff
@@ -893,16 +893,18 @@ def _infer_Gather(self, node):
                                           data_shape[:axis] + indices_shape + data_shape[axis + 1:]))
         # for 1D input, do some sympy compute
         if node.input[0] in self.sympy_data_ and len(data_shape) == 1 and 0 == get_attribute(node, 'axis', 0):
-            idx = self._get_value(node, 1)
-            data = self.sympy_data_[node.input[0]]
-            if type(data) == list:
-                if type(idx) == np.ndarray and len(idx.shape) == 1:
-                    self.sympy_data_[node.output[0]] = [data[int(i)] for i in idx]
+            idx = self._try_get_value(node, 1)
+            if idx is not None:
+                data = self.sympy_data_[node.input[0]]
+                print('data: ', data, ', type(data): ', type(data))
+                if type(data) == list:
+                    if type(idx) == np.ndarray and len(idx.shape) == 1:
+                        self.sympy_data_[node.output[0]] = [data[int(i)] for i in idx]
+                    else:
+                        self.sympy_data_[node.output[0]] = data[int(idx)]
                 else:
```

Comment:
The [onnx spec](https://github.com/onnx/onnx/blob/master/docs/Operators.md#Gather) mentioned that the indices is "Tensor of int32/int64 indices". what situation that it is not list?